src/web/assets/cp/src/js/BaseElementIndex.js

```diff
@@ -909,11 +938,44 @@ Craft.BaseElementIndex = Garnish.Base.extend(
             ? Garnish.$scrollContainer
             : this.$main
           ).scrollTop(0);
-          this._updateView(params, response.data);
+          this._updateView(this.viewParams, response.data);
+
+          if (this.criteriaHasChanged() && !this.sourceHasChanged()) {
+            const itemLabel = this.getItemLabel();
+            const itemsLabel = this.getItemsLabel();
+
+            this._countResults().then((total) => {
+              let successMessage;
+
+              if (!this._isViewPaginated) {
+                successMessage = Craft.t(
+                  'app',
+                  'Showing {total, number} {total, plural, =1{{item}} other{{items}}}',
+                  {
+                    total: total,
+                    item: itemLabel,
+                    items: itemsLabel,
+                  }
+                );
+              } else {
+                const first = this.getFirstItemNumber(total);
+                successMessage = Craft.t(
+                  'app',
+                  'Showing {first, number}-{last, number} of {total, number} {total, plural, =1{{item}} other{{items}}}',
+                  {
+                    first: first,
+                    last: this.getLastItemNumber(first, total),
+                    total: total,
+                    item: itemLabel,
+                    items: itemsLabel,
+                  }
+                );
+              }
 
-          if (pageChanged) {
```

Comment:
@brandonkelly would it be safe to remove this from the function parameters since it's not being used anymore?

Reply:
Sure!